RepoMan/RepoMan/Repository/PullRequestReaderFactory.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using RepoMan.Records;
+using Serilog;
+
+namespace RepoMan.Repository
+{
+    public class PullRequestReaderFactory :
+        IPullRequestReaderFactory
+    {
+        private readonly GitHubPullRequestReaderFactory _ghReaderFactory;
+        private readonly BitBucketCloudPullRequestReaderFactory _bbReaderFactory;
+        private readonly ILogger _logger;
+
+        public PullRequestReaderFactory(GitHubPullRequestReaderFactory ghClientFactory, BitBucketCloudPullRequestReaderFactory bbCloudClientFactory, ILogger logger)
+        {
+            _ghReaderFactory = ghClientFactory ?? throw new ArgumentNullException(nameof(ghClientFactory));
+            _bbReaderFactory = bbCloudClientFactory ?? throw new ArgumentNullException(nameof(bbCloudClientFactory));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public IRepoPullRequestReader GetReader(WatchedRepository repo)
+        {
+            switch (repo.RepositoryKind)
+            {
+                case RepositoryKind.GitHub:
+                case RepositoryKind.GitHubEnterprise:
+                    return _ghReaderFactory.GetReader(repo);
```

Comment:
I don't think there's a functional difference between GH and GHE, is there @bojanrajkovic ? You ran analysis against GHE, but I have not.

Reply:
That's correct, there's no functional difference, as long as GHE is relatively recent.